Created new React Native Components entry

ericsonrd opened the pull request:
<!--
👋 Hi, thanks for sending a PR to Codecademy Docs! 💖
Please fill out all fields below to ensure your PR is reviewed quickly.

**IMPORTANT**

If you would like to receive credit for your contribution to an entry, make sure to have your Codecademy user profile linked to your GitHub account:

1. Go to your Codecademy dashboard.
2. Click on your profile image in the top-right and then choose "Profile".
3. Then click "Edit Profile".
4. In the "GitHub Username" field, add your username (without the @).
5. Click "Save Changes"

Of course, you can opt not to do this and be listed as an "Anonymous contributor", instead. :)
-->

### Description

New entry about React Native Components for the React Native section in the React topic. It gives clear explanation of React Native components, how they relate to React and what they do to generate fast, native-like UI performance.

### Type of Change

<!--- Please delete or cross off the bullet point(s) that are irrelevant to this PR: -->

- Adding a new entry
- Updating the documentation

### Checklist

<!-- Please check ALL the boxes: -->

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

<!--
Having trouble with the PR checker? Here are some common issues and resolutions:

- verify_formatting is failing
  - run `yarn format path/to/markdown/file.md` or `yarn format:all` and commit the results
- verify_lint is failing
  - same as above
  - if verify_lint is still failing, running `yarn lint` locally should let you know what needs to be changed by hand
- test is failing
  - ensure any new markdown files have a `Title` and `Description` defined in their metadata
  - ensure any new markdown files only contain alphanumerics and dashes in their file names and have the same name as their parent directory
  - if that looks ok, running `yarn test` locally should let you know what the issue is
-->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Codecademy/docs?pullRequest=1269) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Codecademy/docs?pullRequest=1269) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Codecademy/docs?pullRequest=1269) it.</sub>

SSwiniarski commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
```diff
@@ -0,0 +1,159 @@
+
+## React Native Components are based on React
+
+![React and React Native Components](../../../../../media/react_react_native_components.png)
```
```suggestion
![React and React Native Components](https://github.com/Codecademy/docs/tree/main/media/react_react_native_components.png)
```

SSwiniarski commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
```diff
@@ -0,0 +1,159 @@
+
+## Native Components
+
+These previously mentioned platform-backed components are called **Native Components,** and React Native creates the corresponding, platform specific Views (whether iOS or Android) for these components at runtime. Because of this, React Native apps look, feel and perform like Native apps.
```
```suggestion
These previously mentioned platform-backed components are called Native Components, and React Native creates the corresponding, platform specific Views (whether iOS or Android) for these components at runtime. Because of this, React Native apps look, feel and perform like Native apps.
```

SSwiniarski commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
```diff
@@ -0,0 +1,159 @@
+
+## Core Components
+
+React Native offers a set of essential, ready-to-use Native Components called **Core Components.** There are many components ranging from Text to Activity Indicators. Most apps will use these Core Components:
```
```suggestion
React Native offers a set of essential, ready-to-use Native Components called Core Components. There are many components ranging from Text to Activity Indicators. Most apps will use these Core Components:
```

SSwiniarski commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
```diff
@@ -0,0 +1,159 @@
+
+## Community Components
+
+React Native Componens can also be custom-built, and there’s a big ecosystem of these Community-built Components that can be accessed on the [**Native Directory**](https://reactnative.directory/)
```
```suggestion
React Native Components can also be custom-built, and there’s a big ecosystem of these Community-built Components that can be accessed on the [Native Directory](https://reactnative.directory/)
```

SSwiniarski commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
```diff
@@ -0,0 +1,159 @@
+
+![React and React Native Components](../../../../../media/react_react_native_components.png)
+
+React Native Components share the same API structure as **React Components.** Whatever a component returns is rendered as a React element, which allows to describe what’s seen on the screen. They can also be defined as ***Function Component*** or ***Class Components:***
```
```suggestion
React Native Components share the same API structure as React Components. Whatever a component returns is rendered as a React element, which allows to describe what’s seen on the screen. They can also be defined as Function Component or Class Components:
```

SSwiniarski commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
```diff
@@ -0,0 +1,159 @@
+
+React Native Components share the same API structure as **React Components.** Whatever a component returns is rendered as a React element, which allows to describe what’s seen on the screen. They can also be defined as ***Function Component*** or ***Class Components:***
+
+ - #### Function Component:
```
```suggestion
 ### Function Component:
```

SSwiniarski commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
````diff
@@ -0,0 +1,159 @@
+React Native Components share the same API structure as **React Components.** Whatever a component returns is rendered as a React element, which allows to describe what’s seen on the screen. They can also be defined as ***Function Component*** or ***Class Components:***
+
+ - #### Function Component:
+	```jsx
````
```suggestion

```jsx
```

SSwiniarski commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
````diff
@@ -0,0 +1,159 @@
+    
+    export default Box;
+    ```
+    Try this code example [HERE.](https://snack.expo.dev/@ericsonrd/rn-function-component)
````
```suggestion

Try this code example [here.](https://snack.expo.dev/@ericsonrd/rn-function-component)
```

SSwiniarski commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
````diff
@@ -0,0 +1,159 @@
+    ```
+    Try this code example [HERE.](https://snack.expo.dev/@ericsonrd/rn-function-component)
+    
+ - #### Class Component:
````
```suggestion
### Class Component:
```

SSwiniarski commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
````diff
@@ -0,0 +1,159 @@
+    Try this code example [HERE.](https://snack.expo.dev/@ericsonrd/rn-function-component)
+    
+ - #### Class Component:
+	```jsx
````
```suggestion

```jsx
```

SSwiniarski commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
````diff
@@ -0,0 +1,159 @@
+    
+    export default Box;
+    ```
+    Try this code example [HERE.](https://snack.expo.dev/@ericsonrd/rn-class-component)
````
```suggestion

Try this code example [here.](https://snack.expo.dev/@ericsonrd/rn-class-component)
```

SSwiniarski commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
```diff
@@ -0,0 +1,159 @@
+
+## JSX, Props and State.
+
+React Native Components also use **JSX,** accept **Props** and manage **State.**
```
```suggestion
React Native Components also use JSX, accept Props and manage State.
```

SSwiniarski commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
```diff
@@ -0,0 +1,159 @@
+
+React Native Components also use **JSX,** accept **Props** and manage **State.**
+
+ - ### JSX
```
```suggestion
### JSX
```

SSwiniarski commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
```diff
@@ -0,0 +1,159 @@
+React Native Components also use **JSX,** accept **Props** and manage **State.**
+
+ - ### JSX
+	As in React, the JSX syntax in React Native allows writing elements inside JavaScript, and also the use of variables inside them:
```
```suggestion

As in React, the JSX syntax in React Native allows writing elements inside JavaScript, and also the use of variables inside them:
```

SSwiniarski commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
````diff
@@ -0,0 +1,159 @@
+ - ### JSX
+	As in React, the JSX syntax in React Native allows writing elements inside JavaScript, and also the use of variables inside them:
+
+	```jsx
````
```suggestion
``jsx
```

SSwiniarski commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
````diff
@@ -0,0 +1,159 @@
+
+    export default Box;
+    ```
+    Try this code example [HERE.](https://snack.expo.dev/@ericsonrd/rn-component-jsx)
````
```suggestion

Try this code example [here.](https://snack.expo.dev/@ericsonrd/rn-component-jsx)
```

SSwiniarski commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
````diff
@@ -0,0 +1,159 @@
+    ```
+    Try this code example [HERE.](https://snack.expo.dev/@ericsonrd/rn-component-jsx)
+	
+ - ### Props
````
```suggestion
### Props
```

SSwiniarski commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
```diff
@@ -0,0 +1,159 @@
+    Try this code example [HERE.](https://snack.expo.dev/@ericsonrd/rn-component-jsx)
+	
+ - ### Props
+	 Most React Native Core Components accept Props, for example, you can pass different sizes via Props to this Box component:
```
```suggestion

Most React Native Core Components accept Props, for example, you can pass different sizes via Props to this Box component:

```

SSwiniarski commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
````diff
@@ -0,0 +1,159 @@
+	
+ - ### Props
+	 Most React Native Core Components accept Props, for example, you can pass different sizes via Props to this Box component:
+	```jsx
````
```suggestion
```jsx
```

SSwiniarski commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
````diff
@@ -0,0 +1,159 @@
+
+    export default BoxCollection;
+    ```
+    Try this code example [HERE.](https://snack.expo.dev/@ericsonrd/rn-component-props)
````
```suggestion

Try this code example [here.](https://snack.expo.dev/@ericsonrd/rn-component-props)
```

SSwiniarski commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
````diff
@@ -0,0 +1,159 @@
+    ```
+    Try this code example [HERE.](https://snack.expo.dev/@ericsonrd/rn-component-props)
+	
+ - ### State
````
```suggestion
### State
```

SSwiniarski commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
```diff
@@ -0,0 +1,159 @@
+    Try this code example [HERE.](https://snack.expo.dev/@ericsonrd/rn-component-props)
+	
+ - ### State
+	 Just like React, React Native Components use State to handle data that changes over time, such as with user interaction:
```
```suggestion

Just like React, React Native Components use State to handle data that changes over time, such as with user interaction:

```

SSwiniarski commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
````diff
@@ -0,0 +1,159 @@
+	
+ - ### State
+	 Just like React, React Native Components use State to handle data that changes over time, such as with user interaction:
+	```js
````
```suggestion
```js
```

SSwiniarski commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
````diff
@@ -0,0 +1,159 @@
+
+    export default Box;
+    ```
+    Try this code example [HERE.](https://snack.expo.dev/@ericsonrd/rn-component-state)
````
```suggestion

Try this code example [here.](https://snack.expo.dev/@ericsonrd/rn-component-state)
```

SSwiniarski reviewed (changes requested):
@ericsonrd, I did a first review and left some comments.

I also want to ask @Dusch4593 how we feel about the off-site links to code.

ericsonrd commented:
> @ericsonrd, I did a first review and left some comments.
> 
> I also want to ask @Dusch4593 how we feel about the off-site links to code.

Got it. Will be checking and solving all these. Also let me know about the off-site links and I'll fix it if neccessary. Thanks for your review.

SSwiniarski reviewed (changes requested):
One more change before passing on to a second review: Go through the file and make sure there are line breaks before and after headers and code blocks.

ericsonrd commented:
> @ericsonrd One more change before passing on to a second review: Go through the file and make sure there are line breaks before and after headers and code blocks.

Understood. Should I leave or remove the external links to code?

SSwiniarski reviewed (approved):
@ericsonrd, looks ready for a second review.

Dusch4593 commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
```diff
@@ -0,0 +1,160 @@
+
+## Community Components
+
+React Native Components can also be custom-built, and there’s a big ecosystem of these Community-built Components that can be accessed on the [Native Directory](https://reactnative.directory/)
```
```suggestion
Views can also be custom-built, and there’s a big ecosystem of these community-built components that can be accessed on the [Native Directory](https://reactnative.directory/).
```

Dusch4593 commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
```diff
@@ -0,0 +1,160 @@
+
+React Native Components can also be custom-built, and there’s a big ecosystem of these Community-built Components that can be accessed on the [Native Directory](https://reactnative.directory/)
+
+## React Native Components are based on React
+
+![React and React Native Components](https://github.com/Codecademy/docs/tree/main/media/react_react_native_components.png)
```
```suggestion
![React and React Native Components](https://github.com/Codecademy/docs/tree/main/media/react_react_native_components.png)
```

Dusch4593 commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
```diff
@@ -0,0 +1,160 @@
+
+![React and React Native Components](https://github.com/Codecademy/docs/tree/main/media/react_react_native_components.png)
+
+React Native Components share the same API structure as React Components. Whatever a component returns is rendered as a React element, which allows to describe what’s seen on the screen. They can also be defined as Function Component or Class Components:
```
```suggestion
### Examples

React Natives uses the same component syntax structure for its views to display elements to the screen, like in React.js. The following examples are of a `Box` component defined as both a class and functional component:
```

Dusch4593 commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
```diff
@@ -0,0 +1,160 @@
+![React and React Native Components](https://github.com/Codecademy/docs/tree/main/media/react_react_native_components.png)
+
+React Native Components share the same API structure as React Components. Whatever a component returns is rendered as a React element, which allows to describe what’s seen on the screen. They can also be defined as Function Component or Class Components:
+
+### Function Component:
+
```
```suggestion

```

Dusch4593 commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
````diff
@@ -0,0 +1,160 @@
+### Function Component:
+
+```jsx
+import React from 'react';
+import { Text } from 'react-native';
+
+const Box= () => {
+    return (
+        <Text>I have a small box</Text>
+    );
+}
+
+export default Box;
````
```suggestion
import React, { Component } from 'react';
import { Text } from 'react-native';

// Functional Component
const Box = () => {
  return (
      <Text>I have a small box</Text>
  );
}

// Class Component
class Box extends Component {
  render() {
      return (
          <Text>I have a small box</Text>
      );
  }
}

export default Box;
```

Dusch4593 commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
````diff
@@ -0,0 +1,160 @@
+export default Box;
+```
+    
+### Class Component:
+
+```jsx
+import React, { Component } from "react";
+import { Text } from 'react-native';
+
+class Box extends Component {
+    render() {
+        return (
+            <Text>I have a small box</Text>
+        );
+    }
+}
+
+export default Box;
+```
````
```suggestion
```

I would replace this with the following **note**:

\> \*\*Note:\*\* To test this example, either the class or function definition of the \`Box\` component must be commented out before doing so.

Dusch4593 commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
```diff
@@ -0,0 +1,160 @@
+	
+## JSX, Props and State.
+
+React Native Components also use JSX, accept Props and manage State.
```
```suggestion
Views also use [JSX](https://www.codecademy.com/resources/docs/react/jsx), accept [props](https://www.codecademy.com/resources/docs/react/props), and manage [state](https://www.codecademy.com/resources/docs/react/state).
```

Dusch4593 commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
````diff
@@ -0,0 +1,160 @@
+As in React, the JSX syntax in React Native allows writing elements inside JavaScript, and also the use of variables inside them:
+
+```jsx
+import React from 'react';
+import { Text } from 'react-native';
+
+const Box= () => {
+    const size = “small”;
+    return (	
+        <Text>I have a {size} box</Text>
+    );
+}
+
+export default Box;
````
```suggestion
import React from 'react';
import { Text } from 'react-native';

const Box = () => {
  const size = “small”;
  return (	
      <Text>I have a {size} box</Text>
  );
}

export default Box;
```

Dusch4593 commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
```diff
@@ -0,0 +1,160 @@
+	
+### Props
+
+Most React Native Core Components accept Props, for example, you can pass different sizes via Props to this Box component:
```
```suggestion
Most Core Components in React Native accept props. For example, different sizes for the `Box` component can be passed via props:
```

Dusch4593 commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
````diff
@@ -0,0 +1,160 @@
+Most React Native Core Components accept Props, for example, you can pass different sizes via Props to this Box component:
+
+```jsx
+import React from 'react';
+import { View, Text } from 'react-native';
+
+const Box = (props) => {
+    return (
+        <Text>I have a {props.size} box</Text>
+    );
+}
+
+const BoxCollection = () => {
+    return (
+        <View>
+            <Box size=“small” />
+            <Box size=“medium” />
+            <Box size=“large” />
+        </View>
+    );
+}
+
+export default BoxCollection;
````
```suggestion
import React from 'react';
import { View, Text } from 'react-native';

const Box = (props) => {
  return (
      <Text>I have a {props.size} box</Text>
  );
}

const BoxCollection = () => {
  return (
      <View>
          <Box size=“small” />
          <Box size=“medium” />
          <Box size=“large” />
      </View>
  );
}

export default BoxCollection;
```

Dusch4593 commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
```diff
@@ -0,0 +1,160 @@
+	
+### State
+
+Just like React, React Native Components use State to handle data that changes over time, such as with user interaction:
```
```suggestion
Like in React, components in React Native also use state to handle data that changes over time, such as with user interaction:
```

Dusch4593 commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
````diff
@@ -0,0 +1,160 @@
+Just like React, React Native Components use State to handle data that changes over time, such as with user interaction:
+
+```jsx
+import React, { useState } from "react";
+import { View, Text, Button } from "react-native";
+
+const Box = () => {
+    const [size, setSize] = useState("small");
+        return (
+            <View>
+                <Text>I have a {size} box </Text>
+                <Button
+                    color="red"
+                    onPress={() => setSize("small")}
+                    title="Small" />
+                <Button
+                    color="blue"
+                    onPress={() => setSize("medium")}
+                    title="Medium" />
+                <Button
+                    color="orange"
+                    onPress={() => setSize("large")}
+                    title="Large" />
+            </View>
+        );
+}
+
+export default Box;
````
```suggestion
import React, { useState } from "react";
import { View, Text, Button } from "react-native";

const Box = () => {
  const [size, setSize] = useState("small");
  return (
    <View>
      <Text>I have a {size} box </Text>
      <Button
        color="red"
        onPress={() => setSize("small")}
        title="Small" />
      <Button
        color="blue"
        onPress={() => setSize("medium")}
        title="Medium" />
      <Button
        color="orange"
        onPress={() => setSize("large")}
        title="Large" />
    </View>
  );
}

export default Box;
```

ericsonrd replied:
Hey @Dusch4593 thanks for the review! I've made the changes but before I update this pull request, I have a question if you don't mind. 

I'm new to Git and Github, and I'm confused by something. For some reason, I don't know if I did something wrong or what, but some files that I didn't created nor modified started showing up when staging files to my forked repository of the Codecademy docs. 

What should I do with this files? I believe if I commit the changes to this React Native entry, these files would appear from my branch and would be merged with this entry, I don't know if those files are up to date. Should I delete these? then commit and update the pull request?

<img width="777" alt="Screenshot 2022-12-18 at 7 43 58 PM" src="https://user-images.githubusercontent.com/95394815/208325986-ed5ef6d1-6170-4e4e-900a-3bf2ac340b1b.png">

ericsonrd replied:
Thanks @Dusch4593 I did that without the --hard flag and It says branch is now updated. This is what I get know. I assume all these files are ignored when I commit rn_components right?

<img width="1054" alt="Screenshot 2022-12-21 at 12 50 27 AM" src="https://user-images.githubusercontent.com/95394815/208826240-bbd1dffa-f261-436d-8e00-b52b61694cae.png">

Dusch4593 replied:
@ericsonrd Cool! Now, can we run `git checkout main` and run `git status` and see what the screenshot shows there?

ericsonrd replied:
@Dusch4593 Before checking out to main, it prompts me to remove these: 
<img width="864" alt="Screenshot 2022-12-26 at 1 07 02 PM" src="https://user-images.githubusercontent.com/95394815/209570475-89c1eb1b-bb30-4dd6-a376-c317279a6ef0.png">

yangc95 replied:
@ericsonrd Try `git stash` and then switch over to the main branch. If I'm trying to maneuver quickly between branches and if the current branch I'm working on has changes on it that I don't want to commit just yet.

ericsonrd replied:
HI @yangc95, thanks for answering. I did that before but the group of files at the bottom still would be overwirtten. Can these be safely removed then checkout?
<img width="889" alt="Screenshot 2022-12-26 at 11 58 14 PM" src="https://user-images.githubusercontent.com/95394815/209608838-58b3ed86-97cd-444b-adc2-bbe1447aeca3.png">

Dusch4593 replied:
@ericsonrd You could try `git stash --include-untracked` and then you should be able to checkout into `main`.

Also, if you want to see the running list of changes you've stashed, run `git stash list`. 😄

ericsonrd replied:
@Dusch4593 Done. It's checked out to `main`. Now what's next? Update `main' from the Codecademy docs then Merge `main` into `rn_components` and continue from there?

<img width="891" alt="Screenshot 2022-12-27 at 8 11 05 PM" src="https://user-images.githubusercontent.com/95394815/209738458-514c8c07-c526-402b-b15c-706c8023e950.png">

Dusch4593 replied:
@ericsonrd Awesome! Now even there it says that remote `main` is in sync with `origin/main`, what happens if we run `git pull`?

ericsonrd replied:
@Dusch4593 Says it's already up to date

Dusch4593 replied:
Gotcha. Just wanted to confirm your local `main` was at least untangled. Now, it looks like your remote `origin/main` may need untangled next:

<img width="903" alt="Screen Shot 2022-12-28 at 6 27 38 PM" src="https://user-images.githubusercontent.com/23144845/209884815-f166256b-538d-44f0-916f-2f8db66fb79f.png">

It is expected for your fork's `origin/main` is x-commits _behind_ the base `Codecademy/docs:main` branch since new changes from outside PR branches are being merged. But sync problems may occur for your fork's `origin/main` if it is x-commits _ahead_, as shown above. A solution that's worked for me is a hard `reset` back to the base `main` branch. On your local `main` branch, assuming you have already have a remote named `upstream` pointing to `https://github.com/Codecademy/docs.git`, let's run the following commands:

```
git reset --hard upstream/main
```

This solution was borrowed from [this StackOverflow thread](https://stackoverflow.com/questions/41283955/github-keeps-saying-this-branch-is-x-commits-ahead-y-commits-behind#:~:text=Now%20to%20fix,match%20upstream%3A)

Give this a try and let me know if you have any issues. 😄

ericsonrd replied:
Done. This is what is shows:

<img width="562" alt="Screenshot 2022-12-28 at 7 56 50 PM" src="https://user-images.githubusercontent.com/95394815/209886566-cdeb3f03-1061-486a-98af-6a355f30e127.png">

Dusch4593 replied:
Ok now try `git pull` again.

ericsonrd replied:
Done. 

<img width="1112" alt="Screenshot 2022-12-28 at 8 11 47 PM" src="https://user-images.githubusercontent.com/95394815/209887330-e15dc99e-4b70-4eec-9a5b-43e0f5257943.png">
<img width="697" alt="Screenshot 2022-12-28 at 8 12 02 PM" src="https://user-images.githubusercontent.com/95394815/209887338-16379560-66e5-4129-921f-408ee45b46b4.png">

Dusch4593 replied:
Hey @ericsonrd! Hoping you had a restful holiday and happy new year! 🎆

I'm sorry that [your fork's `main` branch](https://github.com/ericsonrd/cc-docs-frk/tree/main) still seems to be a bit entangled. After discussing this with a colleague, we were wondering if you'd be open to lifting your fork's branch protections for today to see if we can fix this on our end?

cc: @yangc95

ericsonrd replied:
Hi @Dusch4593! Happy new year to you too!

Done with the branch protections, let me know if it works. I've been reading some guides and now everything is up to date except for the "commits ahead" on main/rn_components, so it seems the next step is Rebase, right?

yangc95 replied:
@Dusch4593 @ericsonrd it looks like permissions are still denied. Let's continue this conversation in https://github.com/Codecademy/docs/discussions/categories/q-a

yangc95 commented on content/react/concepts/react-native/react-native-components/react-native-components.md:
```diff
@@ -0,0 +1,160 @@
+---
+Title: 'React Native Components'
```
@Dusch4593 I'm wondering if maybe we can use the term "Component" instead of views. For React Native it looks like View is one of the many Components that can be utilized

yangc95 replied:
Maybe we can use "Native Components" instead? View is one of the native components so this may cause some confusion. Also, the images added support this term usage and provides the iOS/Android parallel.

Dusch4593 replied:
Totally cool with that 😄

yangc95 commented:
@ericsonrd apologies for any confusion! I mistakenly pushed up some changes but see that you're still working on getting the changes up.

ericsonrd commented:
> @ericsonrd apologies for any confusion! I mistakenly pushed up some changes but see that you're still working on getting the changes up.

Hey @yangc95, @Dusch4593. Added the latest changes to this entry. In the first pharagraph I modified the first description of Components, because it would read "Components are components...", then in the table instead of changing "React Native Components" to "Native Components", changed it to "Core Components" to match the classification on the React Native documentation.

Let me know if it's ok or it would need changes.

yangc95 commented on content/react/concepts/react-native/terms/components/components.md:
```diff
@@ -0,0 +1,153 @@
+  - 'paths/front-end-engineer-career-path'
+---
+
+[**Components**](https://www.codecademy.com/resources/docs/react/components) are units of reusable code that describe the appearance and behavior of the user interface (UI) of a mobile application.
```
```suggestion
[**Components**](https://www.codecademy.com/resources/docs/react/components) are units of reusable code that describe the appearance and behavior of a mobile application's [user interface (UI)](https://www.codecademy.com/resources/docs/uiux/ui-design).
```

yangc95 commented on content/react/concepts/react-native/terms/components/components.md:
```diff
@@ -0,0 +1,153 @@
+
+## Core Components
+
+React Native offers a set of essential, ready-to-use Native Components called Core Components. There are many components ranging from Text to Activity Indicators. Most apps will use these Core Components:
```
```suggestion
React Native offers a set of essential, ready-to-use native components called core components. There are many components ranging from text to activity indicators. Most apps will use these core components:
```

yangc95 commented on content/react/concepts/react-native/terms/components/components.md:
```diff
@@ -0,0 +1,153 @@
+
+## Community Components
+
+Components can also be custom-built, and there’s a big ecosystem of these community-built components that can be accessed on the [Native Directory](https://reactnative.directory/).
```
```suggestion
Components can also be custom-built; there’s a big ecosystem of these community-built components that can be accessed on the [Native Directory](https://reactnative.directory/).
```

yangc95 commented on content/react/concepts/react-native/terms/components/components.md:
```diff
@@ -0,0 +1,153 @@
+
+### Examples
+
+React Native uses the same component syntax structure for its views to display elements to the screen, like in React.js. The following examples are of a `Box` component defined as both a class and functional component:
```
```suggestion
React Native uses the same component syntax structure for its views to display elements to the screen, like in [React.js](https://www.codecademy.com/resources/docs/react). The following examples are of a `Box` component defined as both a class and functional component:
```

yangc95 commented on content/react/concepts/react-native/terms/components/components.md:
```diff
@@ -0,0 +1,153 @@
+
+### JSX
+
+As in React, the JSX syntax in React Native allows writing elements inside JavaScript, and also the use of variables inside them:
```
```suggestion
As in React, the JSX syntax in React Native allows elements and variables to be written inside the JavaScript:
```

yangc95 commented on content/react/concepts/react-native/terms/components/components.md:
```diff
@@ -0,0 +1,153 @@
+	
+### Props
+
+Most Core Components in React Native accept props. For example, different sizes for the `Box` component can be passed via props:
```
```suggestion
Most core components in React Native accept props. For example, different sizes for the `Box` component can be passed via props:
```

yangc95 reviewed (commented):
@ericsonrd I had a few suggestions/comments to add 🙂

Dusch4593 commented on content/react/concepts/react-native/terms/components/components.md:
```diff
@@ -0,0 +1,153 @@
+import { Text } from 'react-native';
+
+// Functional Component
+const Box = () => {
+  return (
+      <Text>I have a small box</Text>
+  );
+}
+
+// Class Component
+class Box extends Component {
+  render() {
+      return (
+          <Text>I have a small box</Text>
+      );
+  }
+}
```
```suggestion
const Box = () => {
  return (
    <Text>I have a small box</Text>
  );
}

// Class Component
class Box extends Component {
  render() {
    return (
        <Text>I have a small box</Text>
    );
  }
}
```

Dusch4593 commented on content/react/concepts/react-native/terms/components/components.md:
```diff
@@ -0,0 +1,153 @@
+
+> **Note:** To test this example, either the class or function definition of the `Box` component must be commented out before doing so.
+
+## JSX, Props and State.
```
```suggestion
## JSX, Props, and State
```

Unless phrased as a question, punctuation shouldn't be at the end of a header.

Dusch4593 commented on content/react/concepts/react-native/terms/components/components.md:
```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import { Text } from 'react-native';
+
+const Box = () => {
+  const size = “small”;
+  return (	
+      <Text>I have a {size} box</Text>
+  );
+}
```
```suggestion
const Box = () => {
  const size = “small”;
  return (	
    <Text>I have a {size} box</Text>
  );
}
```

Dusch4593 commented on content/react/concepts/react-native/terms/components/components.md:
```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import { View, Text } from 'react-native';
+
+const Box = (props) => {
+  return (
+      <Text>I have a {props.size} box</Text>
+  );
+}
+
+const BoxCollection = () => {
+  return (
+      <View>
+          <Box size=“small” />
+          <Box size=“medium” />
+          <Box size=“large” />
+      </View>
+  );
+}
```
```suggestion
const Box = (props) => {
  return (
    <Text>I have a {props.size} box</Text>
  );
}

const BoxCollection = () => {
  return (
    <View>
        <Box size=“small” />
        <Box size=“medium” />
        <Box size=“large” />
    </View>
  );
}
```

Dusch4593 reviewed (changes requested):
@ericsonrd Just some last points of feedback from me. Mostly to do with spacing in the code snippets. 😄

ericsonrd commented:
> @ericsonrd Just some last points of feedback from me. Mostly to do with spacing in the code snippets. 😄

@Dusch4593 Done. 👍🏽

Dusch4593 commented on content/react/concepts/react-native/terms/components/components.md:
```diff
@@ -0,0 +1,153 @@
+| `<View>` | A common container component that supports layout with flexbox, styles, touch handling, accessibility controls, and can contain other components inside such as other views. It is analogous to a non-scrolling [`<div>`](https://www.codecademy.com/resources/docs/html/elements/div) HTML element. | 
+| `<Text>` | Displays text and supports styles and touch events. It is analogous to a [paragraph element](https://www.codecademy.com/resources/docs/html/paragraphs). | 
+| `<Image>` | Displays different types of images, including from network, static, local disks, and from ‘data:’ [URI](https://www.codecademy.com/resources/docs/general/uri) scheme. It is analogous to an [image element](https://www.codecademy.com/resources/docs/html/images). | 
+| `<TextInput>` | Allows the input of text by the user and provides several configuration capabilities such as auto-correction, auto-capitalization, placeholder text, etc. It is analogous to an [`<input>`] element with its `type` attribute set to `"text"`. |
```
```suggestion
| `<TextInput>` | Allows the input of text by the user and provides several configuration capabilities such as auto-correction, auto-capitalization, placeholder text, etc. It is analogous to an [`<input>`](https://www.codecademy.com/resources/docs/html/elements/input) element with its `type` attribute set to `"text"`. |
```

Dusch4593 reviewed (approved):
@ericsonrd Just one last suggestion from me, but looks good otherwise! 😄

Is there anything left to add, change, or check @yangc95?

yangc95 commented on content/react/concepts/react-native/terms/components/components.md:
```diff
@@ -0,0 +1,153 @@
+  const [size, setSize] = useState("small");
+  return (
+    <View>
+      <Text>I have a {size} box </Text>
```
```suggestion
      <Text>I have a {size} box</Text>
```

ericsonrd commented:
Hurray!! 🎉👏🏼👏🏼

Thanks for all your support and patience @Dusch4593 and @yangc95 helping me navigate the "entanglement", I've sure learned a bit about how to manage it better. And glad to have contributed to Codecademy docs!

Be well.

yangc95 commented:
@ericsonrd Thanks for your consistent efforts!

Would you also be able to open a PR for a small change? It looks like some images weren't rendered and need to be replaced with a link that has this format: https://raw.githubusercontent.com/Codecademy/docs/main/media/`filename`

In addition, you're currently a private contributor so you're anonymous on this entry.
If you'd like to make yourself public, add your GitHub account to your Codecademy profile and sign back in with GitHub!
<img width="839" alt="Screen Shot 2023-01-06 at 9 52 27 AM" src="https://user-images.githubusercontent.com/72277593/211037617-74137ad2-2ee3-4f39-8650-ddb7db0d116f.png">

cc: @Dusch4593 @SSwiniarski

ericsonrd commented:
Done @yangc95 and @Dusch4593.

PR Link: https://github.com/Codecademy/docs/pull/1466#issue-1523349506

yangc95 commented:
@ericsonrd Merged! Thanks for making that change ^^
